MailSim/MailSim/ProvidersOM/MailStoreProviderOM.cs

```diff
@@ -189,9 +190,50 @@ public IAddressBook GetGlobalAddressList()
             if (_outlook != null && !_keepOutlookRunning)
             {
                 Console.WriteLine("Exiting Outlook");
- 
+
                 ((Outlook._Application)_outlook).Quit();
             }
         }
+
+
+        /// <summary>
+        /// Finds the Global Address List associated with the MailStore
+        /// </summary>
+        /// <param name="folderPath">parameter for FindFolder</param>
+        /// <returns>IMailFolder for the target of the event</returns>
+        /// In this case the Folder element is a path in Outlook. Each component of the path separated by '\'.
+        /// The first or default folder in the path, can be preceded by "\\" or nothing. If it's the only part of
+        /// the path, then it MUST be one of the default OL folders (see the schema for the EventMonitor operation). 
+        public IMailFolder FindFolder(string folderPath)
+        {
+            IMailFolder folder;
+            string backslash = @"\";
+            try
+            {
+                if (folderPath.StartsWith(@"\\"))
+                {
+                    folderPath = folderPath.Remove(0, 2);
+                }
+                String[] folders =
+                    folderPath.Split(backslash.ToCharArray());
+                folder =
+                    GetDefaultFolder(folders[0]);
+
+                if (folder != null)
+                {
+                    for (int i = 1; i <= folders.GetUpperBound(0); i++)
```

Comment:
You should be able to skip the 1st item, for instance:

foreach (var f in folders.Skip(1))

Let’s just keep this in mind as an option…

Alex

From: tomjebo [mailto:notifications@github.com]
Sent: Wednesday, August 12, 2015 11:12 PM
To: OfficeDev/Interop-MailSim <Interop-MailSim@noreply.github.com>
Cc: Alex Martynov (Wicresoft) <v-am@microsoft.com>
Subject: Re: [Interop-MailSim] IMailStore.FindFolder and EvenMonitor to support path with subfolders (#21)


In MailSim/MailSim/ProvidersOM/MailStoreProviderOM.cs<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fOfficeDev%2fInterop-MailSim%2fpull%2f21%23discussion_r36943451&data=01%7c01%7cv-am%40microsoft.com%7c8850068c46bc494790fe08d2a3a621fd%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=81AgaRyP0bq9i8ozIYPW4hSA%2bkpknZPWYNanCT7fY7E%3d>:

> +            IMailFolder folder;

> +            string backslash = @"\";

> +            try

> +            {

> +                if (folderPath.StartsWith(@"\\"))

> +                {

> +                    folderPath = folderPath.Remove(0, 2);

> +                }

> +                String[] folders =

> +                    folderPath.Split(backslash.ToCharArray());

> +                folder =

> +                    GetDefaultFolder(folders[0]);

> +

> +                if (folder != null)

> +                {

> +                    for (int i = 1; i <= folders.GetUpperBound(0); i++)

.Length is fine since folders is a simple one dimension array. I will use that.
However, foreach will go through all the elements. In this case, we want to skip the first element. So we should keep the for(...) construct.

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fOfficeDev%2fInterop-MailSim%2fpull%2f21%2ffiles%23r36943451&data=01%7c01%7cv-am%40microsoft.com%7c8850068c46bc494790fe08d2a3a621fd%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=nu8lywsR9SeNRm1ZC8S0xm3%2fZ%2bqgLhoXLCmNobovL%2fU%3d>.